debian/fop.1
.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.47.3.
.TH FOP "1" "March 2016" "fop Version 2.1" "User Commands"
.SH NAME

fop \- Command line wrapper for the java version of fop.
.SH DESCRIPTION
Apache FOP (Formatting Objects Processor) is a print formatter driven by XSL
formatting objects (XSL-FO) and an output independent formatter. It is a Java
application that reads a formatting object (FO) tree and renders the resulting
pages to a specified output. Output formats currently supported include PDF,
PS, PCL, AFP, XML (area tree representation), Print, AWT and PNG, and to a
lesser extent, RTF and TXT. The primary output target is PDF.
.PP
.SH SYNOPSIS

fop [options] [\-fo|\-xml] infile [\-xsl file] [\-awt|\-pdf|\-mif|\-rtf|\-tiff|\-png|\-pcl|\-ps|\-txt|\-at [mime]|\-print] <outfile>
.IP
.SH OPTIONS
.TP
\fB\-version\fR
print FOP version and exit
.TP
\fB\-x\fR
dump configuration settings
.TP
\fB\-c\fR cfg.xml
use additional configuration file cfg.xml
.TP
\fB\-l\fR lang
the language to use for user information
.TP
\fB\-nocs\fR
disable complex script features
.TP
\fB\-r\fR
relaxed/less strict validation (where available)
.TP
\fB\-dpi\fR xxx
target resolution in dots per inch (dpi) where xxx is a number
.TP
\fB\-s\fR
for area tree XML, down to block areas only
.TP
\fB\-v\fR
run in verbose mode (currently simply print FOP version and continue)
.TP
\fB\-o\fR [password]
PDF file will be encrypted with option owner password
.TP
\fB\-u\fR [password]
PDF file will be encrypted with option user password
.TP
\fB\-noprint\fR
PDF file will be encrypted without printing permission
.TP
\fB\-nocopy\fR
PDF file will be encrypted without copy content permission
.TP
\fB\-noedit\fR
PDF file will be encrypted without edit content permission
.TP
\fB\-noannotations\fR
PDF file will be encrypted without edit annotation permission
.TP
\fB\-nofillinforms\fR
PDF file will be encrypted without fill in interactive form fields permission
.TP
\fB\-noaccesscontent\fR
PDF file will be encrypted without extract text and graphics permission
.TP
\fB\-noassembledoc\fR
PDF file will be encrypted without assemble the document permission
.TP
\fB\-noprinthq\fR
PDF file will be encrypted without print high quality permission
.TP
\fB\-a\fR
enables accessibility features (Tagged PDF etc., default off)
.TP
\fB\-pdfprofile\fR prof
PDF file will be generated with the specified profile
(Examples for prof: PDF/A\-1b or PDF/X\-3:2003)
.TP
\fB\-conserve\fR
enable memory\-conservation policy (trades memory\-consumption for disk I/O)
(Note: currently only influences whether the area tree is serialized.)
.TP
\fB\-cache\fR
specifies a file/directory path location for the font cache file
.TP
\fB\-flush\fR
flushes the current font cache file
.IP
.SH INPUT
.TP
infile
xsl:fo input file (the same as the next)
(use '\-' for infile to pipe input from stdin)
.TP
\fB\-fo\fR
infile       xsl:fo input file
.TP
\fB\-xml\fR infile
xml input file, must be used together with \fB\-xsl\fR
.TP
\fB\-atin\fR infile
area tree input file
.TP
\fB\-ifin\fR infile
intermediate format input file
.TP
\fB\-imagein\fR infile
image input file (piping through stdin not supported)
.TP
\fB\-xsl\fR stylesheet
xslt stylesheet
.TP
\fB\-param\fR name value <value> to use for parameter <name> in xslt stylesheet
(repeat '\-param name value' for each parameter)
.TP
\fB\-catalog\fR
use catalog resolver for input XML and XSLT files
.IP
.SH OUTPUT
.TP
outfile
input will be rendered as PDF into outfile
(use '\-' for outfile to pipe output to stdout)
.TP
\fB\-pdf\fR outfile
input will be rendered as PDF (outfile req'd)
.TP
\fB\-pdfa1b\fR outfile
input will be rendered as PDF/A\-1b compliant PDF
(outfile req'd, same as "\-pdf outfile \fB\-pdfprofile\fR PDF/A\-1b")
.TP
\fB\-awt\fR
input will be displayed on screen
.TP
\fB\-rtf\fR outfile
input will be rendered as RTF (outfile req'd)
.TP
\fB\-pcl\fR outfile
input will be rendered as PCL (outfile req'd)
.TP
\fB\-ps\fR outfile
input will be rendered as PostScript (outfile req'd)
.TP
\fB\-afp\fR outfile
input will be rendered as AFP (outfile req'd)
.TP
\fB\-tiff\fR outfile
input will be rendered as TIFF (outfile req'd)
.TP
\fB\-png\fR outfile
input will be rendered as PNG (outfile req'd)
.TP
\fB\-txt\fR outfile
input will be rendered as plain text (outfile req'd)
.TP
\fB\-at\fR [mime] out
representation of area tree as XML (outfile req'd)
specify optional mime output to allow the AT to be converted
to final format later
.TP
\fB\-if\fR [mime] out
representation of document in intermediate format XML (outfile req'd)
specify optional mime output to allow the IF to be converted
to final format later
.TP
\fB\-print\fR
input file will be rendered and sent to the printer
see options with "\-print help"
.TP
\fB\-out\fR mime outfile input will be rendered using the given MIME type
(outfile req'd) Example: "\-out application/pdf D:\eout.pdf"
(Tip: "\-out list" prints the list of supported MIME types and exits)
.TP
\fB\-svg\fR outfile
input will be rendered as an SVG slides file (outfile req'd)
Experimental feature \- requires additional fop\-sandbox.jar.
.TP
\fB\-foout\fR outfile
input will only be XSL transformed. The intermediate
XSL\-FO file is saved and no rendering is performed.
(Only available if you use \fB\-xml\fR and \fB\-xsl\fR parameters)
.IP
.SH EXAMPLES
.IP

fop foo.fo foo.pdf

fop \fB\-fo\fR foo.fo \fB\-pdf\fR foo.pdf (does the same as the previous line)

fop \fB\-xml\fR foo.xml \fB\-xsl\fR foo.xsl \fB\-pdf\fR foo.pdf

fop \fB\-xml\fR foo.xml \fB\-xsl\fR foo.xsl \fB\-foout\fR foo.fo

fop \fB\-xml\fR \- \fB\-xsl\fR foo.xsl \fB\-pdf\fR \-

fop foo.fo \fB\-mif\fR foo.mif

fop foo.fo \fB\-rtf\fR foo.rtf

fop foo.fo \fB\-print\fR

fop foo.fo \fB\-awt\fR
.SH HYPENATION

fop can make use of external hyphenation rules that can be supplied in
the form of a \fBhyphenation jar\fR. You can find such a one at
\fIhttp://offo.sourceforge.net/\fR.

To actually use that, set the \fBFOP_HYPHENATION_PATH\fR environment variable
to the full path of this file, or use configuration files, see below.
.SH "CONFIGURATION FILES"

fop reads two configuration files on startup: \fI/etc/fop.conf\fR and
\fI$HOME/.foprc\fR. These are plain shell script files where you can define
for instance the \fBFOP_HYPHENATION_PATH\fR variable.

Another way to extend is \fIconf.d/\fR files. Simply add a configuration file
into \fI/etc/fop.conf.d/my.conf\fR.
.SH "JAVA-WRAPPERS NOTE"
This program is a shell script wrapper based on \fBjava-wrappers(7)\fR. You
therefore benefit from several features; please see the \fBjava-wrappers(7)\fR
manual page for more information about them.
.SH AUTHOR
Wrapper author: Ola Lundqvist <opal@debian.org>

This manual page was written by Mathieu Malaterre <malat@debian.org>,
for the Debian project (and may be used by others).
.SH "SEE ALSO"
.BR fop-ttfreader(1),
.BR java-wrappers(7).


debian/changelog
fop (1:2.1-7) unstable; urgency=medium

  * Team upload.
  * Fixed the build failure caused by the latest qdox update (Closes: #871106)

 -- Emmanuel Bourg <ebourg@apache.org>  Sun, 06 Aug 2017 15:36:06 +0200

fop (1:2.1-6) unstable; urgency=high

  * Team upload.
  * Fixed CVE-2017-5661: Information disclosure vulnerability (Closes: #860567)

 -- Emmanuel Bourg <ebourg@apache.org>  Wed, 24 May 2017 15:53:03 +0200

fop (1:2.1-5) unstable; urgency=medium

  * Fix FTBFS now that comments are parsed by javadoc. Closes: #852910
    + debian/patches/javadoc_allow-script-in-comments.patch

 -- Mathieu Malaterre <malat@debian.org>  Mon, 30 Jan 2017 11:36:02 +0100

fop (1:2.1-4) unstable; urgency=medium

  * Team upload.
  * Transition to the Servlet API 3.1

 -- Emmanuel Bourg <ebourg@apache.org>  Mon, 19 Sep 2016 10:06:08 +0200

fop (1:2.1-3) unstable; urgency=medium

  [ Alexis Bienvenüe ]
  * Use d/changelog date for pod2man
  * Strip user/host from MANIFEST
    + debian/patches/strip_user_host_from_manifest.patch
  * Fix javadoc encoding. Closes: #820684
    + debian/patches/javadoc_encoding.patch

  [ Mathieu Malaterre ]
  * Update offo to release 2.2. Closes: #817264
  * Bump Std-Vers to 3.9.8, no changes needed

 -- Mathieu Malaterre <malat@debian.org>  Fri, 15 Apr 2016 16:33:18 +0200

fop (1:2.1-2) unstable; urgency=medium

  [ Sylvestre Ledru ]
  * Remove myself from the uploaders

  [ Mathieu Malaterre ]
  * Fix Couldn't find hyphenation pattern. Closes: #805472 

 -- Mathieu Malaterre <malat@debian.org>  Mon, 15 Feb 2016 08:47:55 +0100

fop (1:2.1-1) unstable; urgency=medium

  * New upstream release. Closes: #813572
  * Remove patch applied upstream:
    - debian/patches/bug799662.patch

 -- Mathieu Malaterre <malat@debian.org>  Mon, 08 Feb 2016 18:51:00 +0100

fop (1:2.0+dfsg-5) unstable; urgency=medium

  * Team upload.
  * Fixed the groupId of the avalon artifacts (Closes: #804564)

 -- Emmanuel Bourg <ebourg@apache.org>  Tue, 05 Jan 2016 15:35:09 +0100

fop (1:2.0+dfsg-4) unstable; urgency=medium

  * allow overriding logging options. Closes: #717320
  * Fix Null pointer exception. Closes: #799662

 -- Mathieu Malaterre <malat@debian.org>  Fri, 06 Nov 2015 20:29:38 +0100

fop (1:2.0+dfsg-3) unstable; urgency=medium

  * Upload to sid
  * Update d/watch to check signature.

 -- Mathieu Malaterre <malat@debian.org>  Sun, 13 Sep 2015 14:02:43 +0200

fop (1:2.0+dfsg-2) experimental; urgency=medium

  * Fix logging (default to info level).
  * Refresh patches (DEP-3 header)

 -- Mathieu Malaterre <malat@debian.org>  Sun, 16 Aug 2015 13:55:19 +0200

fop (1:2.0+dfsg-1) experimental; urgency=medium

  * New upstream. Closes: #792132
  * Transition to libservlet3.0-java. Closes: #709178 
  * Bump Std-Vers to 3.9.6, no changes needed
  * Remove old Breaks/Replace (oldstable version)
  * Update d/watch to use d/copyright File-Excluded
  * Fix some lintian warnings

 -- Mathieu Malaterre <malat@debian.org>  Thu, 16 Jul 2015 21:55:30 +0200

fop (1:1.1.dfsg2-2) unstable; urgency=medium

  * Fix hyphenation issue. Closes: #780579

 -- Mathieu Malaterre <malat@debian.org>  Sat, 11 Jul 2015 21:25:11 +0200

fop (1:1.1.dfsg2-1) unstable; urgency=medium

  * Team upload.
  * Removed the non-free sRGB color profile and use the equivalent profile
    from the icc-profiles-free package (Closes: #657281)

 -- Emmanuel Bourg <ebourg@apache.org>  Sun, 02 Nov 2014 17:20:55 +0100

fop (1:1.1.dfsg-2) unstable; urgency=low

  * Upload to sid
  * Make sure fop install correct libfop-java version
  * fop now required qdox >= 1.12 to build
  * Bump Std-Vers to 3.9.4, no changes needed
  * Remove servlet-api from fop.sh. Closes: #671273
  * Use newer d/watch (HowToHelpWithFixingWatchFiles)

 -- Mathieu Malaterre <malat@debian.org>  Thu, 16 May 2013 10:39:12 +0200

fop (1:1.1.dfsg-1) experimental; urgency=low

  * New upstream. Closes: #691474
  * Use my @d.o alias, remove DMUA flags
  * Bump Std-Vers to 3.9.3, no changes needed
  * Remove patches applied upstream:
    - d/p/enco.patch
    - d/p/enco2.patch
    - d/p/fixnpeinlinestacking.patch
    - d/p/illegalargex.patch
  * d/control: now requires xmlgraphics-commons >= 1.5

 -- Mathieu Malaterre <malat@debian.org>  Sat, 03 Nov 2012 15:35:23 +0100

fop (1:1.0.dfsg2-6) unstable; urgency=low

  [ Mathieu Malaterre ]
  * Support conf.d like configuration files. Closes: #630198
  * Fix fop man page (outdated). Closes: #654592, #512443
  * Split packages into fop and libfop-java. Closes: #544546

  [ Joachim Breitner ]
  * Include hyphenation patterns from TeX. Closes: #655027

 -- Mathieu Malaterre <mathieu.malaterre@gmail.com>  Sun, 15 Jan 2012 11:13:23 +0100

fop (1:1.0.dfsg2-5) unstable; urgency=low

  * Need ant 1.8.2 at least update d/control.
  * Fix NPE in inline element. Closes: #649476
  * Fix IllegalArgumentException. Closes: #646141

 -- Mathieu Malaterre <mathieu.malaterre@gmail.com>  Tue, 03 Jan 2012 14:44:06 +0100

fop (1:1.0.dfsg2-4) unstable; urgency=low

  * Migrate fop to use libservlet2.5-java. Closes: #646088
  * Updating the fop Uploaders list. Closes: #654018
  * Update VCS urls.

 -- Mathieu Malaterre <mathieu.malaterre@gmail.com>  Mon, 02 Jan 2012 11:12:37 +0100

fop (1:1.0.dfsg2-3) unstable; urgency=low

  * Fix misrender issue. Closes: #610344
  * Remove qdox.jar copy from source tree
  * Properly fix logging option. Closes: #575411

 -- Mathieu Malaterre <mathieu.malaterre@gmail.com>  Sun, 11 Sep 2011 15:01:03 +0200

fop (1:1.0.dfsg2-2) unstable; urgency=low

  [ Mathieu Malaterre ]
  * Fix issue with qdox. Closes: #639162
    - Also dump exception for verbosity

  [ Miguel Landaeta ]
  * Add mh_clean call in clean target. (Closes: #639515).

 -- Mathieu Malaterre <mathieu.malaterre@gmail.com>  Thu, 25 Aug 2011 15:35:55 +0200

fop (1:1.0.dfsg2-1) unstable; urgency=low

  [ Sylvestre Ledru ]
  * Upload to unstable
  * Dependency on libxp removed (Closes: #623639)
  * Standards-Version updated to version 3.9.2

debian/fop.sh
#! /bin/sh

# Shell script wrapper around the fop program,
# Copyright 2008 by Vincent Fourmond <fourmond@debian.org>
#
# Licensed under the same terms as fop itself, that is under
# the conditions of the Apache 2 licence.

# Include the wrappers utility script
. /usr/lib/java-wrappers/java-wrappers.sh

# comment this line if you want fop to run without headless property,
# or write a line containing
#  HEADLESS=
# in your fop configuration file.
HEADLESS=-Djava.awt.headless=true

# the following lines are copied from upstream `fop` wrapper:
#
# The default commons logger for JDK1.4 is JDK1.4Logger.
# To use a different logger, uncomment the one desired below
# LOGCHOICE=-Dorg.apache.commons.logging.Log=org.apache.commons.logging.impl.NoOpLog
LOGCHOICE=-Dorg.apache.commons.logging.Log=org.apache.commons.logging.impl.SimpleLog
# LOGCHOICE=-Dorg.apache.commons.logging.Log=org.apache.commons.logging.impl.Log4JLogger

# Logging levels
# Below option is only if you are using SimpleLog instead of the default JDK1.4 Logger.
# To set logging levels for JDK 1.4 Logger, edit the %JAVA_HOME%/JRE/LIB/logging.properties 
# file instead.
# Possible SimpleLog values:  "trace", "debug", "info" (default), "warn", "error", or "fatal".
LOGLEVEL=-Dorg.apache.commons.logging.simplelog.defaultlog=INFO

for cf in /etc/fop.conf.d/*.conf; do
    if [ -f $cf ]; then
        . $cf;
    fi
done

# Load system-wide configuration, if any
if [ -f /etc/fop.conf ]; then
    . /etc/fop.conf
fi

# Load user's preferences, if any
if [ -f "$HOME/.foprc" ]; then
    . $HOME/.foprc
fi

# We prefer to use openjdk or Sun's java if available
find_java_runtime openjdk sun  || find_java_runtime 

find_jars commons-io avalon-framework serializer xalan2 xml-apis 
find_jars batik-all commons-logging xercesImpl xmlgraphics-commons
find_jars xml-apis-ext fontbox

# We load the hyphenation jar at the request of the user.
if [ "$FOP_HYPHENATION_PATH" ]; then
    find_jars $FOP_HYPHENATION_PATH
fi
find_jars /usr/share/fop/fop-hyph.jar
find_jars fop

run_java $HEADLESS $LOGCHOICE $LOGLEVEL org.apache.fop.cli.Main "$@"
